shell/platform/embedder/embedder_surface_gl.cc

```diff
@@ -80,6 +80,16 @@ EmbedderSurfaceGL::GLProcResolver EmbedderSurfaceGL::GetGLProcResolver() const {
   return gl_dispatch_table_.gl_proc_resolver;
 }
 
+// |GPUSurfaceGLDelegate|
+SurfaceFrame::FramebufferInfo EmbedderSurfaceGL::GLContextFramebufferInfo()
+    const {
+  // Enable partial repaint by default on the embedders.
+  auto info = SurfaceFrame::FramebufferInfo{};
+  info.supports_readback = true;
+  info.supports_partial_repaint = true;
+  return info;
+}
```

Comment:
This looks god to me. The only thing is that the Embedder API was trying to disable partial repaint in cases that do not support it to prevent ABI breakages. For example, if an Embedder has not defined a populate_existing_callback (which is likely since it is a new callback), partial repaint should be disabled. However, in this case, partial repaint might be triggered unnecessarily. In practice, I believe that it will not always be triggered because the check to run it also verifies that there is existing damage (https://github.com/flutter/engine/blob/main/shell/common/rasterizer.cc#:~:text=if%20(frame%2D%3Eframebuffer_info().existing_damage,damage%2D%3ESetPreviousLayerTree(last_layer_tree_.get())%3B). However, I think it would still be nice to have the Embedder have the option to disable partial repaint whenever the given inputs are not valid (be it callbacks, the format of the passed existing damage, and whatnot).

Reply:
Fixed it by only setting it to true if the callback is provided.